src/components/global/Navbar.jsx

```diff
@@ -1,8 +1,147 @@
-const Navbar = () => {
-  // any logic needed
-  // return (
-  // all the JSX (looks like HTML) to render
-  // )
+import {
+  AppBar,
+  Box,
+  Hidden,
+  IconButton,
+  List,
+  ListItem,
+  Popover,
+  Toolbar,
+  Typography
+} from '@mui/material';
+import MenuIcon from '@mui/icons-material/Menu';
+import { Brightness4, Brightness7 } from '@mui/icons-material';
+import { BrowserRouter as Route, Link } from 'react-router-dom';
+import { useState, useRef } from 'react';
+
+const logoBlobStyle = {
+  position: 'absolute',
+  bottom: 0,
+  left: -30,
+  right: 0,
+  top: -50,
+  height: '250px',
+  backgroundImage: { md: 'url(/assets/logoBlobNav.svg)', sm: 'none' },
+  backgroundRepeat: 'no-repeat',
+  display: {
+    lg: 'flex',
+    xs: 'none'
+  },
+  maxWidth: '200px'
+};
+
+const textStyle = {
+  textDecoration: 'none',
+  color: 'inherit'
 };
 
-export default Navbar;
+function NavBar() {
+  const [darkMode, setDarkMode] = useState(false);
+  const [menuOpen, setMenuOpen] = useState(false);
+
+  // anchor for menu to pop from its icon
+  const anchorRef = useRef(null);
+
+  const handleThemeChange = () => {
```

Comment:
To the best of my knowledge react has batch state updates and using  the format ` setMenuOpen(!menuOpen)` could cause some issues if its spammed, this is more of an issue if multiple async values can come in and change state. Not a big deal since its only light/dark mode so either would work here.

Reply:
> To the best of my knowledge react has batch state updates and using the format ` setMenuOpen(!menuOpen)` could cause some issues if its spammed, this is more of an issue if multiple async values can come in and change state. Not a big deal since its only light/dark mode so either would work here.

ahh yes, gotcha.